app/models/user.rb

```diff
@@ -4,21 +4,24 @@ class User < ActiveRecord::Base
   devise :database_authenticatable, :registerable, :confirmable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  has_many :pending_purchases
+  has_many :line_items
+  has_one :cart
+
+  after_save :create_cart
+
+  def create_cart
+    Cart.create!(user: self)
+  end
 
   def add_to_cart(book)
```

Comment:
You're totally right that the user is taking the action, and that's still true since the user has the method on it.This way it's just delegating the internal behavior of what happens to a book in a cart to the cart itself (which I would guess is why Brandon had you add a cart class in the first place).